stonesoup/models/transition/orbital/orbit.py

```diff
@@ -370,6 +371,205 @@ def transition(self, orbital_state, time_interval=timedelta(seconds=0)):
+            The time interval over which to test the new state (default is 1
+            second)
+
+        Returns
+        -------
+        : CovarianceMatrix
+            The transition covariance matrix
+
+        """
+        return CovarianceMatrix(self.noise * time_interval.total_seconds())
+
+    def function(self, orbital_state, noise=0,
+                 time_interval=timedelta(seconds=0)):
+        r"""Just passes parameters to the transition function
+
+        Parameters
+        ----------
+        orbital_state : :class:`~.OrbitalState`
+            The prior orbital state
+        noise : CovarianceMatrix, optional
+            The nominal covariance matrix. This isn't passed to the transition
+            function though, so can be left out.
+        time_interval: :math:`\delta t` :attr:`datetime.timedelta`, optional
+            The time interval over which to test the new state (default is 0
+            seconds)
+
+        Returns
+        -------
+        : StateVector
+            The orbital state vector returned by the transition function
+
+        Note
+        ----
+        This merely passes the parameters to the :attr:`.transition()`
+        function. The noise parameter has no effect, is included merely
+        for compatibility with parent classes.
+        """
+        return self.transition(orbital_state, time_interval)
+
+    def transition(self, orbital_state, time_interval=timedelta(seconds=0)):
+        r"""The transition proceeds as algorithm 3.4 in [1]
+
+        Parameters
+        ----------
+        orbital_state : :class:`~.OrbitalState`
+            The prior orbital state
+        time_interval: :math:`\delta t` :attr:`datetime.timedelta`, optional
+            The time interval over which to test the new state (default
+            is 0 seconds)
+
+        Returns
+        -------
+        : StateVector
+            The orbital state vector returned by the transition function
+
+        Warning
+        -------
+        If noisy samples from the transition function are required, use
+        the :attr:`rvs` method.
+        """
+        # Get the position and velocity vectors
+        # Evaluated at initial timestamp
+        tle_ext = Satrec.twoline2rv(orbital_state.state.metadata['line_1'],
+                                    orbital_state.state.metadata['line_2'])
+        jd, fr = jday(orbital_state.state.timestamp.year,
+                      orbital_state.state.timestamp.month,
+                      orbital_state.state.timestamp.day,
+                      orbital_state.state.timestamp.hour,
+                      orbital_state.state.timestamp.minute,
+                      orbital_state.state.timestamp.second)
+        e, bold_r_0, bold_v_0 = tle_ext.sgp4(jd, fr)
+
+        tt = orbital_state.state.timestamp + time_interval
+        jd, fr = jday(tt.year,tt.month,tt.day,
+                      tt.hour,tt.minute,tt.second)
+        e, bold_r, bold_v = tle_ext.sgp4(jd, fr)
```

Comment:
Warning here. These values are in km and km/s. OrbitalState does not require specification of particular units and so will break if used with (e.g.) a method which is initialised with other units. 

At least need to include a comment. Better would be some sort of checking...

Reply:
Further, default initialisation of OrbitalState assumes units in m.

Reply:
Good spot. I should probably add another test to help detect incorrect units.